ocaml/networkd/lib/network_utils.ml

```diff
@@ -198,7 +198,9 @@ module Sysfs = struct
         false
 
   let list () =
-    let is_dir name = Sys.is_directory ("/sys/class/net/" ^ name) in
+    let is_dir name =
+      try Sys.is_directory ("/sys/class/net/" ^ name) with _ -> false
+    in
     Sys.readdir "/sys/class/net"
     |> Array.to_list
     |> List.filter (fun name -> is_dir name && is_ether_device name)
```

Comment:
An alternative would be to check using `Sys.file_exists` in the predicate in addition. `file_exists` is true for directories, too.

Reply:
I think here we want _only_ directories.

Reply:
My suggestion was to check for: exists && is directory && is_ether_devive.

Reply:
Oh you said _in addition_ (to the directory check)...